vbeam/core/probe_geometry.py

```diff
@@ -0,0 +1,109 @@
+"""A datastructure for representing a transducer probe excluding 
+the position, orientation, etc.
+"""
+
+from typing import Callable, Optional
+from vbeam.fastmath import numpy as np
+from vbeam.fastmath.traceable import traceable_dataclass
+from vbeam.util.geometry.v2 import distance
+
+identity_fn = lambda x: x  # Just return value as-is
+
+
+@traceable_dataclass(("ROC", "rx_aperture_length_s", "tx_aperture_length_s"))
+class ProbeGeometry:
+    """A container for probe parameters.
+    The aperture lengths are arc lenghts of the probe surface in azimuth and elevation. 
+    """
+    ROC: tuple[float, float] # (azimuth, elevation)
+    rx_aperture_length_s: tuple[float, float] = None # (width, height)
+    tx_aperture_length_s: tuple[float, float] = None # (width, height)
+    def __postinit__(self):           
+        if type(self.ROC) is not tuple or len(self.ROC) != 2:
+            self.ROC = (self.ROC, self.ROC)
+    def __getitem__(self, *args) -> "ProbeGeometry":
+        _maybe_getitem = lambda attr: attr.__getitem__(*args) if attr is not None else None
+        return self.ProbeGeometry(
+            _maybe_getitem(self.ROC),
+            _maybe_getitem(self.rx_aperture_length_s),
+            _maybe_getitem(self.tx_aperture_length_s),
+        )
+
+    @property
+    def curvature_center(self):
+        return (np.array([0.0,0.0,-self.ROC[0]]), np.array([0.0,0.0,-self.ROC[1]]) ) 
+    
+    @property
+    def get_sender_normal(self):
+        vector = self.sender_position - self.curvature_center
+        return vector / distance(vector)
+    
+    @property
+    def get_receiver_normal(self):
+        vector = self.receiver_position - self.curvature_center
+        return vector / distance(vector)
+    
+    def get_theta(self,position):
+        return np.arctan2(position[0],self.ROC[0]+position[2])
+    
+    def get_phi(self,position):
+        return np.arctan2(position[1],self.ROC[1]+position[2])
+    
+    def get_surface_length(self, position):
+        theta = self.get_theta(position=position)
+        phi = self.get_phi(position=position)
+        return (self.ROC*theta, self.ROC*phi)
```

Comment:
What does `position` represent in `get_surface_length`? I'm not 100% following what it means in `get_theta` and `get_phi` either.